### PR TITLE
Implement a few legoomni/mxomni functions

### PR DESCRIPTION
**Pull request by MishaProductions**
- Implemented LegoOmni::PickROI (doesnt match)
 - Implemented LegoOmni::Destroy (doesn't match as not all classes are implemented)
 - Implmented LegoOmni::Notify (100% match)
 - Implemented MxOmni::CreatePresenter (not matching due to something wrong about notification types?)
 - Implemented MxStreamController::FUN_100c1e70 (100% match)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -224,6 +224,7 @@ add_library(lego1 SHARED
   LEGO1/towtrackmissionstate.cpp
   LEGO1/viewmanager/viewmanager.cpp
   LEGO1/viewmanager/viewroi.cpp
+  LEGO1/lego3dview.cpp
```
Comment (foxtacles) [inline CMakeLists.txt:227]: This should be sorted alphabetically with the other units

File: LEGO1/legoomni.cpp [hunk 2]
```diff
@@ -227,13 +229,15 @@ void PlayMusic(MxU32 p_index)
...
 {
 	// TODO
+
+	// This function seems to populate an unknown structure, and then calls 0x1001b230
 }
 
-// STUB: LEGO1 0x1003dd70
-LegoROI* PickROI(MxLong, MxLong)
+// FUNCTION: LEGO1 0x1003dd70
+LegoROI* PickROI(MxLong p_a, MxLong p_b)
```
Comment (foxtacles) [inline LEGO1/legoomni.cpp:237]: This function exists twice? Same address in `Lego3DView`

File: LEGO1/legoomni.cpp [hunk 3]
```diff
@@ -527,11 +604,14 @@ MxBool LegoOmni::DoesEntityExist(MxDSAction& p_dsAction)
 	return FALSE;
 }
 
-// STUB: LEGO1 0x1005b400
-int LegoOmni::GetCurrPathInfo(LegoPathBoundary**, int&)
+// FUNCTION: LEGO1 0x1005b400
+int LegoOmni::GetCurrPathInfo(LegoPathBoundary** p_path, int& p_value)
```
Comment (foxtacles) [inline LEGO1/legoomni.cpp:608]: `int` should be changed to `MxS32`

File: LEGO1/legoworld.cpp [hunk 4]
```diff
@@ -91,6 +91,13 @@ void LegoWorld::FUN_1001fc80(IslePathActor* p_actor)
 {
 }
 
+// STUB: LEGO1 0x10020120
+int LegoWorld::GetCurrPathInfo(LegoPathBoundary** p_path, int& p_value)
```
Comment (foxtacles) [inline LEGO1/legoworld.cpp:95]: Should have `MxS32`

File: LEGO1/mxstreamcontroller.cpp [hunk 5]
```diff
@@ -133,6 +133,21 @@ MxResult MxStreamController::VTable0x30(MxDSAction* p_action)
 	return result;
 }
 
+// FUNCTION: LEGO1 0x100c1e70
+MxCore* MxStreamController::FUN_100c1e70(MxDSAction& p_obj)
```
Comment (foxtacles) [inline LEGO1/mxstreamcontroller.cpp:137]: Would call this parameter `p_action`

File: LEGO1/mxstreamcontroller.cpp [hunk 6]
```diff
@@ -133,6 +133,21 @@ MxResult MxStreamController::VTable0x30(MxDSAction* p_action)
...
+{
+	MxAutoLocker locker(&m_criticalSection);
+	MxCore* result = NULL;
+	if (p_obj.GetObjectId() != -1) {
+		MxDSAction* action = m_unk0x3c.Find(&p_obj, FALSE);
+		if (action != NULL) {
+			result = (MxCore*) action->GetUnknown28();
```
Comment (foxtacles) [inline LEGO1/mxstreamcontroller.cpp:144]: Looks like we should change the type of `m_unk0x28` and the return type of this function to `MxPresenter*`? Based on how this function is used.

File: LEGO1/lego3dmanager.h [hunk 7]
```diff
@@ -10,7 +10,7 @@ class Lego3DManager {
 private:
 	int m_unk0x00;
```
Comment (ItzSwirlz) [inline LEGO1/lego3dmanager.h:11]: Switch to undefined4 type (and for below)

File: LEGO1/mxdsobject.h [hunk 8]
```diff
@@ -38,23 +40,25 @@ class MxDSObject : public MxCore {
...
-	char* m_sourceName;   // 0x10
-	undefined4 m_unk0x14; // 0x14
-	char* m_objectName;   // 0x18
-	MxU32 m_objectId;     // 0x1c
-	MxAtomId m_atomId;    // 0x20
-	MxS16 m_unk0x24;      // 0x24
-	undefined4 m_unk0x28; // 0x28
+	MxU32 m_sizeOnDisk;     // 0x8
+	MxU16 m_type;           // 0xc
+	char* m_sourceName;     // 0x10
```
Comment (ItzSwirlz) [inline LEGO1/mxdsobject.h:55]: Is this confirmed to be a char* or is it a MxChar*

File: LEGO1/mxstreamcontroller.cpp [hunk 9]
```diff
@@ -133,6 +133,21 @@ MxResult MxStreamController::VTable0x30(MxDSAction* p_action)
 	return result;
 }
 
+// FUNCTION: LEGO1 0x100c1e70
+MxCore* MxStreamController::FUN_100c1e70(MxDSAction& p_action)
+{
+	MxAutoLocker locker(&m_criticalSection);
+	MxCore* result = NULL;
+	if (p_action.GetObjectId() != -1) {
```
Comment (ItzSwirlz) [inline LEGO1/mxstreamcontroller.cpp:141]: Is there a constant/define here to replace -1 for something like `NONE`?

File: LEGO1/mxstreamer.cpp [hunk 10]
```diff
@@ -143,6 +143,27 @@ MxResult MxStreamer::FUN_100b99b0(MxDSAction* p_action)
...
+	else
+	{
+		tempAction.SetObjectId(p_dsAction->GetObjectId());
+		tempAction.SetAtomId(p_dsAction->GetAtomId());
+		tempAction.SetUnknown24(p_dsAction->GetUnknown24());
+	}
+
+
+	// TODO: remove action from list
+	return -1;
```
Comment (ItzSwirlz) [inline LEGO1/mxstreamer.cpp:161]:
```suggestion
	return FAILURE;
```

File: LEGO1/mxdsobject.h (hunk 8 above)
Comment (MishaProductions) [reply LEGO1/mxdsobject.h:55]: there isn't an MxChar type added in mxtypes.h, but I would assume that exists

Comment (ItzSwirlz) [reply LEGO1/mxdsobject.h:55]: We got:
```
typedef unsigned char MxU8;
typedef signed char MxS8;
```

**Comment by foxtacles** [comment]
Please do not add additional functions/functionality after opening a PR. This makes reviews a lot harder.

**Comment by foxtacles** [comment]
The notification classes, as you pointed out, still have issues but the rest looks good to me.

**Pull request merged**